src/middleware.js

```diff
@@ -80,11 +86,26 @@ function domainChanged(domain) {
     const timeSpentOnDomain = ((Date.now() - data['openedTime'] - data['totalInactiveTime']) / 1000);
     const dateString = new Date(Date.now()).toLocaleDateString();
 
-    // set default value to an empty object
+    // add data to the map
     addElement(dateString, lastDomain, timeSpentOnDomain);
 
-    // update the last domain
-    setLastDomain(domain);
+    // check if the current domain is in the whitelist
+    chrome.storage.sync.get(["whitelist"], function (data) {
+      var list = data['whitelist'];
+      if (list === undefined) {
+        list = []
+      }
+      if (list.includes(domain)) {
+        if (debugMode) {
+          console.log("White list url found: " + domain);
+        }
+        // reset LastDomain object
+        cleanUsage();
```

Comment:
Why do we clean usage if the list does not contain the domain? Wouldn't this reset all the data?

Reply:
The cleanUsage() method only cleans the 'lastDomain' data, so it will not count time towards the previously visited sites.
For example: Suppose we visited the following sites in order:
             youtube:10min -> google(in white list):20min -> amazon:10min
By cleaning the lastDomain data, we won't append 20 min to youtube's time.
When we visit amazon, since lastDomain was reset, it will not record time on google.